store/principal.go

```diff
@@ -86,13 +86,18 @@ type UpdateUserMessage struct {
 
 // UserMessage is the message for an user.
 type UserMessage struct {
-	ID            int
-	Email         string
-	Name          string
-	Type          api.PrincipalType
-	PasswordHash  string
-	Role          api.Role
-	MemberDeleted bool
+	ID                 int
+	Email              string
+	Name               string
+	Type               api.PrincipalType
+	PasswordHash       string
+	IdentityProviderID *int
```

Comment:
Better to use the resource_id of identity provider. When this is empty string, it's Bytebase auth provider.

Reply:
The means we need to do some SQL joins.